src/gc.c

```diff
@@ -2913,8 +2892,15 @@ JL_DLLEXPORT void jl_gc_collect(int full)
 {
     jl_ptls_t ptls = jl_get_ptls_states();
     if (jl_gc_disable_counter) {
-        gc_num.deferred_alloc += (gc_num.allocd + gc_num.interval);
-        gc_num.allocd = -(int64_t)gc_num.interval;
+        size_t allocd = 0;
+        for (int i = 0; i < jl_n_threads; i++) {
+            jl_ptls_t ptls2 = jl_all_tls_states[i];
```

Comment:
you aren't sync'd here with the other threads--it's invalid to write through `jl_all_tls_states`